### PR TITLE
Fixed refresh create order page on backend side

### PR DESCRIPTION
**Pull request by ananth-iyer**
### Description (*)

This PR will fix Admin create order page is when refreshed then not showing customer selection grid

### Fixed Issues (if relevant)

1. magento/magento2https://github.com/magento/magento2/issues/21603: Adminhtml - Create order - no customer selected
2. ...

### Manual testing scenarios (*)

1. Admin > Sales > Create a new order
2. Then on Customer selection grid - Refresh this page now
3. You will see Customer selection grid again

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php [hunk 1]
```diff
@@ -120,7 +120,7 @@ public function getSaveUrl()
     public function getCustomerSelectorDisplay()
     {
         $customerId = $this->getCustomerId();
-        if ($customerId === null) {
+        if (empty($customerId)) {
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php]: I guess the usage of `empty()` function is redundant in this case, considering the variable always exists.
The issue caused by zero `0` value that does not match the condition.
So, I would recommend simplifying this condition to:
```
if (!$customerId) {
```

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php [hunk 2]
```diff
@@ -135,7 +135,7 @@ public function getStoreSelectorDisplay()
     {
         $storeId = $this->getStoreId();
         $customerId = $this->getCustomerId();
-        if ($customerId !== null && !$storeId) {
+        if (!empty($customerId) && !$storeId) {
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php]: I would recommend simplifying this condition to:
```
if ($customerId && !$storeId) {
```

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php [hunk 3]
```diff
@@ -182,7 +182,7 @@ public function getOrderDataJson()
                 );
             }
         }
-        if ($this->getStoreId() !== null) {
+        if (!empty($this->getStoreId())) {
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php]: I would recommend simplifying this condition to:
```
if ($this->getStoreId()) {
```

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Header.php [hunk 4]
```diff
@@ -76,7 +76,7 @@ protected function _getCreateOrderTitle()
         $customerId = $this->getCustomerId();
         $storeId = $this->getStoreId();
         $out = '';
-        if ($customerId && $storeId) {
+        if (!empty($customerId) && $storeId) {
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Block/Adminhtml/Order/Create/Header.php]: I think, the usage of `!empty` a condition is also redundant here.

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create.php [hunk 5]
```diff
@@ -60,7 +60,7 @@ protected function _construct()
         $this->buttonList->update('save', 'data_attribute', []);
 
         $this->buttonList->update('save', 'id', 'submit_order_top_button');
-        if ($customerId === null || !$storeId) {
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Block/Adminhtml/Order/Create.php]: I would recommend simplifying this condition to:
```
if (!$customerId || !$storeId) {
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hi @ananth-iyer,
thank you for your contribution!

Could you please check the code review recommendations and apply the minor changes in the way I described in a couple of examples.

Thank you!

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create.php (hunk 5 above)
Comment (ananth-iyer) [reply app/code/Magento/Sales/Block/Adminhtml/Order/Create.php]: I have refactored the if condition

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Header.php (hunk 4 above)
Comment (ananth-iyer) [reply app/code/Magento/Sales/Block/Adminhtml/Order/Create/Header.php]: I have refactored the if condition

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php (hunk 3 above)
Comment (ananth-iyer) [reply app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php]: I have refactored the if condition

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php (hunk 2 above)
Comment (ananth-iyer) [reply app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php]: I have refactored the if condition

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php (hunk 1 above)
Comment (ananth-iyer) [reply app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php]: I have refactored the if condition

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create.php (hunk 5 above)
Comment (ananth-iyer) [reply app/code/Magento/Sales/Block/Adminhtml/Order/Create.php]: Please review https://github.com/magento/magento2/pull/21665/commits/9181dc351fd476e695d9425ae1941ace972c18b7 changes

Comment (ananth-iyer) [reply app/code/Magento/Sales/Block/Adminhtml/Order/Create.php]: @dmytro-ch  Please review https://github.com/magento/magento2/pull/21665/commits/9181dc351fd476e695d9425ae1941ace972c18b7 changes

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-4473](https://jira.corp.magento.com/browse/ENGCOM-4473) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Comment by sdzhepa** [comment]
Hello @ananth-iyer 

Thank you for contribution and collaboration!

During testing, we faced an issue.
Could you please have a look and fix it?

The fix affects Reorder flow

- **Problem:** _After click Reorder button on placed order as Guest Customer(Admin Reorder) page for create/select new customer is opened_

**Manual testing scenario:**
1. Go to Storefront
2. Add product to cart
3. Place order as Guest Customer
4. Go to Admin side: SALES > Orders
5. Find in Grid newly placed order and click "View" to open Order Review Page 
6. Click "Reorder"

**Actual Result:** Not OK. Page for select or create new Customer is opened
 
PR_21662_ActualResult



**Expected Result:** Behaviour for this flow should be the same as on `2.3-develop` branch. In other words, fix should not change Reorder flow, it should fix only Refresh page issue https://github.com/magento/magento2/issues/21603

Example of the current `2.3-develop` behavior for Reorder flow:

PR_21662_ExpecetdResult

**Comment by ananth-iyer** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer, here is your [new Magento instance](https://pr-21665.instances.magento-community.engineering).
Admin access: [https://pr-21665.instances.magento-community.engineering/admin](https://pr-21665.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by ananth-iyer** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer. Thank you for your request. I'm working on Magento instance for you

**Comment by ananth-iyer** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer, here is your [new Magento instance](https://pr-21665.instances.magento-community.engineering).
Admin access: [https://pr-21665.instances.magento-community.engineering/admin](https://pr-21665.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by ananth-iyer** [comment]
@sdzhepa  I have found a new issue on customer selection grid:
- There is ajax call for `card_validation` that create a quote before selecting a customer, though it also calls after selecting a customer.
- Here quote is created & saved in the quote session.
- Then the page refreshed:
 You will get customer id = 0 from quote session. This is treated as a guest order.

Issue_21603_Refresh_New_Order_Page

Note: I will fix this in my next commits.

**Comment by ananth-iyer** [comment]
@sdzhepa @pjw345 I was having issue in creating order on multi-store & today I have resolved it. I will push changes after final testing.

**Comment by ananth-iyer** [comment]
@dmytro-ch Please review my latest committed changes and let us all know

**Comment by ananth-iyer** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer, here is your [new Magento instance](https://pr-21665.instances.magento-community.engineering).
Admin access: [https://pr-21665.instances.magento-community.engineering/admin](https://pr-21665.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

File: app/code/Magento/Sales/view/adminhtml/templates/order/create/billing/method/form.phtml [hunk 6]
```diff
@@ -56,11 +56,14 @@
             'Magento_Sales/order/create/form'
         ], function(mage) {
             mage.apply();
-        <?php if ($_methodsCount !== 1) : ?>
-            order.setPaymentMethod('<?= $block->escapeHtml($currentSelectedMethod); ?>');
-        <?php else : ?>
-            payment.switchMethod('<?= $block->escapeHtml($currentSelectedMethod); ?>');
-        <?php endif; ?>
+            if (window.is_order_new_loaded) {
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/view/adminhtml/templates/order/create/billing/method/form.phtml]: Please use "camel" case for properties and variables.
E.g.
```
window.isNewOrderLoaded
```

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php [hunk 7]
```diff
@@ -182,7 +182,7 @@ public function getOrderDataJson()
                 );
             }
         }
-        if ($this->getStoreId() !== null) {
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php]: I'm not sure we really need to remove this strict condition?
Could you please double check if it was not removed accidentally?

**Review by dmytro-ch — CHANGES_REQUESTED** [review]

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php (hunk 7 above)
Comment (ananth-iyer) [reply app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php]: I have checked in single store it selects frontend default store and in multi store it asks to select frontend store then quote is created with selected store id.
So I thought to keep the strict condition.

File: app/code/Magento/Sales/view/adminhtml/templates/order/create/billing/method/form.phtml (hunk 6 above)
Comment (anant-svc) [reply app/code/Magento/Sales/view/adminhtml/templates/order/create/billing/method/form.phtml]: I have changed "is_new_order_loaded" to "isNewOrderLoaded"

File: app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php (hunk 7 above)
Comment (anant-svc) [reply app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php]: I have reverted to restrict condition in latest commit, please verify.

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-4473](https://jira.corp.magento.com/browse/ENGCOM-4473) has been created to process this Pull Request

**Comment by soleksii** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @stoleksiy. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @stoleksiy, here is your [Magento instance](https://i-21665-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21665-2-3-develop.instances.magento-community.engineering/admin](https://i-21665-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by soleksii** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @stoleksiy. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @stoleksiy, here is your [new Magento instance](https://pr-21665.instances.magento-community.engineering).
Admin access: [https://pr-21665.instances.magento-community.engineering/admin](https://pr-21665.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Before:**
before

**After:**
after1

after2

**Comment by VladimirZaets** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets. Thank you for your request. I'm working on Magento instance for you

**Comment by VladimirZaets** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, here is your [new Magento instance](https://pr-21665.instances.magento-community.engineering).
Admin access: [https://pr-21665.instances.magento-community.engineering/admin](https://pr-21665.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by VladimirZaets** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, here is your [Magento instance](https://i-21665-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21665-2-3-develop.instances.magento-community.engineering/admin](https://i-21665-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by VladimirZaets** [comment]
Hi @ananth-iyer, thanks for collaboration. In the current case, your solution isn't the best because you fix can provide additional side effects. 
The main problem in this issue is request in the loadArea method in `Sales/view/adminhtml/web/order/create/scripts.js` because this request required to pass `customer_id` param, otherwise this method redirect to create new customer_id (the case that we have). So, I should check the existence of `customer_id` and if it not exists don't make the request.

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @ananth-iyer, thanks for collaboration. In the current case, your solution isn't the best because you fix can provide additional side effects. 
The main problem in this issue is request in the loadArea method in `Sales/view/adminhtml/web/order/create/scripts.js` because this request required to pass `customer_id` param, otherwise this method redirect to create new customer_id (the case that we have). So, I should check the existence of `customer_id` and if it not exists don't make the request.

**Comment by ananth-iyer** [comment]
@VladimirZaets I have applied your suggested fixes and pushed in the latest commit.
Please review it here: https://github.com/magento/magento2/pull/21665/commits/96b78825b2df0a846df9a4401c490f4456eb5519

**Comment by VladimirZaets** [comment]
@ananth-iyer thanks

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Comment by p-bystritsky** [comment]
@ananth-iyer AdminSubmitsOrderWithAndWithoutFieldsValidationTest is failing. Please, update your PR.

**Comment by ananth-iyer** [comment]
@p-bystritsky Can you please guide me to update my PR?

**Comment by p-bystritsky** [comment]
@ananth-iyer I can only provide you steps to reproduce:
1. Create a Product.
2. Create an Order.
3. Create a new Customer with a length of First Name more than 255 symbols. ("Please enter less or equal than 255 symbols." error message should appear)
4. Edit First Name.
5. Place the Order.

Expected result:
Order placed.

Actual result:
Errors occurred.

Please see a video:

**Comment by ananth-iyer** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer, here is your [new Magento instance](https://pr-21665.instances.magento-community.engineering).
Admin access: [https://pr-21665.instances.magento-community.engineering/admin](https://pr-21665.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by ananth-iyer** [comment]
@p-bystritsky I have updated my branch. then used your reproduce steps and found the issue is gone.
Please test it on your side once.

**Comment by ananth-iyer** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ananth-iyer, here is your [Magento instance](https://i-21665-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21665-2-3-develop.instances.magento-community.engineering/admin](https://i-21665-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-4473](https://jira.corp.magento.com/browse/ENGCOM-4473) has been created to process this Pull Request

**Comment by soleksii** [comment]
Hi @ananth-iyer !

I've checked PR changes and looks like, bug providing by @p-bystritsky  in comment https://github.com/magento/magento2/pull/21665#issuecomment-489590564 is still reproducible!

could you take a look please?

**Comment by sidolov** [comment]
@ananth-iyer , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**